Core/Rewriters/FieldRewriter.cs

```diff
@@ -0,0 +1,216 @@
+﻿//  Copyright (c) rubicon IT GmbH
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated documentation files (the "Software"),
+// to deal in the Software without restriction, including without limitation the rights to use, copy, modify, merge, publish, distribute, sublicense,
+// and/or sell copies of the Software, and to permit persons to whom the Software is furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+// FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY,
+// WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+//
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using RhinoMocksToMoqRewriter.Core.Extensions;
+
+namespace RhinoMocksToMoqRewriter.Core.Rewriters
+{
+  public class FieldRewriter : RewriterBase
+  {
+    private readonly IFormatter _formatter;
+    private List<IFieldSymbol>? _generateMockFieldSymbols;
+
+    public FieldRewriter (IFormatter formatter)
+    {
+      _formatter = formatter;
+    }
+
+    public override SyntaxNode? Visit (SyntaxNode? node)
+    {
+      if (node == null)
+      {
+        return node;
+      }
+
+      try
+      {
+        _generateMockFieldSymbols = new List<IFieldSymbol>();
+        _generateMockFieldSymbols.AddRange (GetFieldSymbols (node));
+        return base.Visit (node);
+      }
+      finally
+      {
+        _generateMockFieldSymbols = null;
+      }
+    }
+
+    public override SyntaxNode? VisitFieldDeclaration (FieldDeclarationSyntax node)
+    {
+      var fieldSymbols = node.Declaration.Variables
+          .Select (v => Model.GetDeclaredSymbol (v) as IFieldSymbol)
+          .ToList();
+
+      if (fieldSymbols.Any (s => s is null))
+      {
+        throw new InvalidOperationException ("Unable to get FieldSymbol from FieldDeclaration");
+      }
+
+      if (_generateMockFieldSymbols == null)
+      {
+        throw new InvalidOperationException ("FieldSymbolList must not be null!");
+      }
+
+      foreach (var fieldSymbol in fieldSymbols)
+      {
+        if (!_generateMockFieldSymbols.Any (s => SymbolEqualityComparer.Default.Equals (s, fieldSymbol)))
+        {
+          return node;
+        }
+      }
```

Comment:
Is this the right thing to do? return if e.g. the first fieldsymbol could not be found in the list? shouldn't all be checked here?